src/ol/interaction/Snap.js

```diff
@@ -68,8 +69,6 @@ function getFeatureFromEvent(evt) {
   }
 }
 
-const tempSegment = [];
```

Comment:
This is not global, instead it is local to this module.  Declaring and assigning it here avoids the unnecessary allocation on every iteration in the loops below.

Reply:
Ok, got it, I will drop first commit